wallet/public/index.html

```diff
@@ -60,19 +37,87 @@
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
     <title>Agoric Wallet</title>
-  </head>
-  <body style="overflow-x: hidden; overflow-y: scroll">
-    <noscript>You need to enable JavaScript to run this app.</noscript>
-    <div id="root"></div>
-    <!--
-      This HTML file is a template.
-      If you open it directly in the browser, you will see an empty page.
+    <script type="module">
+      const MAX_ATTEMPTS = 10;
+      const WAIT_PER_ATTEMPT = 10;
+      let attempts = 0;
+      const startWhenKeplrAvailable = () => {
+        console.debug('looking for keplr');
+        attempts += 1;
+        if (window.keplr) {
+          // XXX copied from lockdown.ts; could be automated
+          // Allow the React dev environment to extend the console for debugging
+          // features.
+          // @ts-expect-error template
+          // eslint-disable-next-line no-constant-condition
+          const consoleTaming =
+            '%NODE_ENV%' === 'development' ? 'unsafe' : 'safe';
+          // @ts-expect-error template
+          // eslint-disable-next-line no-constant-condition
+          const errorTaming =
+            '%NODE_ENV%' === 'development' ? 'unsafe' : 'safe';
 
-      You can add webfonts, meta tags, or analytics to this file.
-      The build step will place the bundled scripts into the <body> tag.
+          // eslint-disable-next-line no-restricted-properties
+          const { pow: mathPow } = Math;
+          // eslint-disable-next-line no-restricted-properties
+          Math.pow = (base, exp) =>
+            typeof base === 'bigint' && typeof exp === 'bigint'
+              ? base ** exp
+              : mathPow(base, exp);
```

Comment:
I realize this was present before, but why are we replacing `Math.pow` with a function that violates the spec? What code are we using that leverages this extension?

Reply:
Yea I'm not sure either, I'm guessing something with `AmountMath` breaks without this when dealing with `bigint`s

Reply:
I don't see _any_ actual use of `Math.pow` in agoric-sdk (or in endo, for that matter):
```sh
$ git log --decorate -1 && git grep -C2 '\bpow\b' packages
commit 51ee28ffe0395ed9a8924388e46b898b6564cf24 (HEAD -> master, upstream/master, origin/master, origin/HEAD)
Merge: 72b0aa9ce 82d1fb540
Author: Brian Warner <warner@lothar.com>
Date:   Fri Jun 2 21:29:03 2023 +0000

    Merge pull request #7880 from Agoric/merge-mainnet1B-rc2-back-into-trunk
    
    merge mainnet1B-rc2 back into trunk
packages/swingset-runner/demo/vatStore2/xorshift128.js-  };
packages/swingset-runner/demo/vatStore2/xorshift128.js-
packages/swingset-runner/demo/vatStore2/xorshift128.js:  // Math.pow(2, -32) = 2.3283064365386963e-10
packages/swingset-runner/demo/vatStore2/xorshift128.js:  // Math.pow(2, -52) = 2.220446049250313e-16
packages/swingset-runner/demo/vatStore2/xorshift128.js-  const random = () => {
packages/swingset-runner/demo/vatStore2/xorshift128.js-    advance();
```

Reply:
Agree. Sometimes we will replace a primordial binding before lockdown in order to repair either a safety problem or a spec non-conformance problem. This one introduces a spec non-conformance problem, and should be deleted if possible.

Curious about the history of this? Seems to have been introduced at https://github.com/Agoric/dapp-card-store/pull/35 but the PR comment does not explain why. Are there others?

Too late for this PR though. File an issue?

Reply:
https://github.com/Agoric/wallet-app/issues/84